README.md

```diff
@@ -135,6 +135,26 @@ If you are using public cloud, then make sure you see the notes from the Rancher
 
 k3s docs: [k3s configuration / open ports](https://rancher.com/docs/k3s/latest/en/configuration/#open-ports-network-security)
 
+## Working with encrypted ssh keys
+
+If the ssh-key is encrypted the first step is to try to connect to the ssh-agent. If this works, it will be used to connect to the server.
+If the ssh-agent is not running, the user will be prompted for the password of the ssh-key.
+
+On most Linux systems and MacOS, ssh-agent is automatically configured and executed at login. No additional actions are required to use it. 
+
+To start the ssh-agent manually and add your key run the following commands:
+
+```
+eval `ssh-agent`
```

Comment:
What does `eval` do?

Reply:
The output of `ssh-agent` is something like this:
```
SSH_AUTH_SOCK=/tmp/ssh-FaHVps3SpMii/agent.3315; export SSH_AUTH_SOCK;
SSH_AGENT_PID=3316; export SSH_AGENT_PID;
echo Agent pid 3316;
```

So `eval`, in this case, sets all the required environment variables.

Reply:
OK. I don't think I've used that tool directly. Maybe indirectly.

So every time someone wants to run k3sup, they have to run the agent first and this command, otherwise they just enter the password manually in the prompt?

What about during automation tasks?

Reply:
> So every time someone wants to run k3sup, they have to run the agent first and this command, otherwise they just enter the password manually in the prompt?

Almost every desktop environment that i am aware of like kde, gnome, xfce starts an ssh-agent on startup and adds your default ssh key to it. No additonal action is required and everything just works.

I'm not sure about macos, though.
I found: https://apple.stackexchange.com/questions/48502/how-can-i-permanently-add-my-ssh-private-key-to-keychain-so-it-is-automatically

> On OSX, the native ssh-add client has a special argument to save the private key's passphrase in the OSX keychain, which means that your normal login will unlock it for use with ssh. On OSX Sierra and later, you also need to configure SSH to always use the keychain